modin/error_message.py

```diff
@@ -36,7 +36,9 @@ def not_implemented(cls, message: str = "") -> NoReturn:
         )
 
     @classmethod
-    def single_warning(cls, message: str) -> None:
+    def single_warning(
+        cls, message: str, category: Union[type[Warning], None] = None
+    ) -> None:
         message_hash = hash(message)
```

Comment:
I think we need to explicitly add a comment that there should not be identical messages with different categories, so only the message is used as the hash key.

Reply:
added